eZ/Publish/Core/REST/Server/Controller/User.php

```diff
@@ -179,16 +179,31 @@ public function loadUser($userId)
         $user = $this->userService->loadUser($userId);
 
         $userContentInfo = $user->getVersionInfo()->getContentInfo();
-        $userMainLocation = $this->locationService->loadLocation($userContentInfo->mainLocationId);
         $contentType = $this->contentTypeService->loadContentType($userContentInfo->contentTypeId);
 
+        $mainLocationId = $userContentInfo->mainLocationId;
+
+        // special case for loading current user, which does not have permissions to load its location.
+        if ($user == $this->repository->getCurrentUser()) {
+            $locationService = $this->locationService;
+            $userMainLocation = $this->repository->sudo(
+                function () use ($locationService, $mainLocationId) {
+                    return $locationService->loadLocation($mainLocationId);
+                }
+            );
+            $relations = array();
```

Comment:
Not really certain how relations should be handled.

If loading the own user's location shouldn't pose any problem, using sudo to load relations could expose restricted content, hence a potential security issue?